server/rest/src/test/java/org/infinispan/rest/resources/CacheV2ResourceTest.java

```diff
@@ -595,6 +598,36 @@ public void testGetProtoCacheConfig() {
       testGetProtoCacheConfig(APPLICATION_JSON_TYPE);
    }
 
+   @Test
+   public void testRebalancingActions() {
+      String cacheName = "default";
+      assertRebalancingStatus(cacheName, true);
+
+      RestCacheClient cacheClient = client.cache(cacheName);
+      RestResponse response = join(cacheClient.disableRebalancing());
+      ResponseAssertion.assertThat(response).isOk();
+      assertRebalancingStatus(cacheName, false);
+
+      response = join(cacheClient.enableRebalancing());
+      ResponseAssertion.assertThat(response).isOk();
+      assertRebalancingStatus(cacheName, true);
+   }
+
+   private void assertRebalancingStatus(String cacheName, boolean enabled) {
+      Security.doAs(ADMIN_USER, () -> {
```

Comment:
I'd rather use `TestingUtil.extractGlobalComponent()` and avoid explicit security code in the test.

Reply:
Updated.